README.md

```diff
@@ -1,111 +1,108 @@
+```
 
-### Network interface mapping
+For ARM64 / aarch64 based systems:
 
-Network interface mapping is required in order for `nm-configurator`
-to identify the proper configurations for each host it is running on.
+```shell
+$ curl -o nmc -L https://github.com/suse-edge/nm-configurator/releases/latest/download/nmc-aarch64
+$ chmod +x nmc
+```
 
-This additional config must be provided in a YAML format mapping the logical name of the interface to its MAC address:
+### Manual method
 
-```yaml
-host_config:
-  - hostname: node1.example.com
-    interfaces:
-      - logical_name: eth0
-        mac_address: 00:10:20:30:40:50
-      - logical_name: eth1
-        mac_address: 10:20:30:40:50:60        
-  - hostname: node2.example.com
-    interfaces:
-      - logical_name: eth0
-        mac_address: 00:11:22:33:44:55
+```shell
+$ git clone https://github.com/suse-edge/nm-configurator.git
+$ cd nm-configurator
+$ cargo build --release # optionally specify --target flag if cross compiling
 ```
 
-**NOTE:** Interface names during the installation of nodes might differ from the preconfigured logical ones.
-This is expected and `nm-configurator` will rely on the MAC addresses and use the actual names for the
-NetworkManager configurations instead e.g. settings for interface with a predefined logical name `eth0` but
-actually named `eth0.101` will automatically be adjusted and stored to `/etc/NetworkManager/eth0.101.nmconnection`.
+## How to run it?
 
-## How to install it?
+### Generate config
 
-### Standard method:
+NMC depends on having the desired network state for all known nodes beforehand.
 
-Each release is published with `nm-configurator` already built for `amd64` and `arm64` Linux systems:
+[NetworkManager](https://documentation.suse.com/sle-micro/5.5/html/SLE-Micro-all/cha-nm-configuration.html)
+is using connection profiles defined as files stored under `/etc/NetworkManager/system-connections`.
+In order to generate these config (*.nmconnection) files, NMC uses the
+[nmstate](https://github.com/nmstate/nmstate) library and requires a configuration directory as an input.
+This directory must contain the desired network state for all hosts in a <i>hostname</i>.yaml file format.
+Please refer to the official nmstate docs for [examples](https://nmstate.io/examples.html).
 
-For AMD64 / x86_64 based systems:
 ```shell
-$ curl -o nm-configurator -L https://github.com/suse-edge/nm-configurator/releases/latest/download/nm-configurator-amd64 
-$ chmod +x nm-configurator
+$ export RUST_LOG=info # set log level ("error" by default)
+$ ./nmc generate --config-dir config --output-dir _out
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node1.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node2.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node3.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc] Successfully generated and stored network config
```

Comment:
I can't seem to get this to work as expected, that's why I think that we need a simple example:
```
rdo@tw-aarch64-build:~/nmc-example> tree config/
config/
├── host_config.yaml
├── node1.example.com
├── node2.example.com
└── node3.example.com

1 directory, 4 files

rdo@tw-aarch64-build:~/nmc-example> cat config/host_config.yaml
- hostname: node1.example.com
  interfaces:
    - logical_name: eth0
      mac_address: FE:C4:05:42:8B:AA
- hostname: node2.example.com
  interfaces:
    - logical_name: eth0
      mac_address: FE:C4:05:42:8B:AB
- hostname: node3.example.com
  interfaces:
    - logical_name: eth0
      mac_address: FE:C4:05:42:8B:AC

rdo@tw-aarch64-build:~/nmc-example> cat config/node1.example.com
interfaces:
- name: eth0
  type: ethernet
  state: up
  ipv4:
    address:
    - ip: 192.168.122.250
      prefix-length: 24
    enabled: true
  ipv6:
    address:
    - ip: 2001:db8::1:1
      prefix-length: 64
    enabled: true

rdo@tw-aarch64-build:~/nmc-example> ./nmc generate --config-dir config --output-dir _out
INFO[2023-11-11T12:13:37Z] starting network manager configurator...
FATA[2023-11-11T12:13:37Z] failed to load static host configuration: yaml: unmarshal errors:
  line 1: cannot unmarshal !!seq into config.Config

```

Reply:
Just drop the `host_config.yaml` file. It is not expected to be created manually.

Reply:
Actually, you're also running the old Go version one. As mentioned, we haven't produced a tag yet so the v0.1.0 release contains the Go binaries.